src/components/AuthorizationHOC.jsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+const AuthorizationHOC = ({ component: Component, token, ...rest }) => (
+  <Route
+    {...rest}
+    render={() => {
+      if (token) {
```

Comment:
@TheDrizzyWay Good job, From the implementation, there is only a check for if the token exists, is there a way of verifying from this client side that the token is valid or expired? If not I think it would be a good condition to add to the HOC, what do you think?

Reply:
Yeah, i guess we could create a utility function for that.